airflow-core/src/airflow/ui/src/layouts/Details/Graph/utils.ts

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import type { Edge } from "@xyflow/react";
+
+type FilterType = "all" | "both" | "downstream" | "upstream";
+
+type GraphParams<T extends { id: string }> = {
+  edges: Array<Edge>;
+  filter: FilterType;
+  nodes: Array<T>;
+  taskId?: string;
+};
+
+export const filterGraph = <T extends { id: string }>({
```

Comment:
We already have `includeDownstream, includeUpstream, root` in `useStructureServiceStructureData` let's use those instead of filtering everything on the frontend